cscs-checks/microbenchmarks/mpi/src/halo_cell_exchange.c

```diff
@@ -1,156 +1,332 @@
+/* This benchmark emulates a halo cell exchange in n dimensions. The pure
+   communication is considered without any stencil computation.*/
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <mpi.h>
 
-int main(int argc, const char* argv[]){
-    MPI_Comm cart_comm, red_comm;
-    MPI_Request *request;
-    MPI_Status *status;
-    int ndims, reorder, color, end, ierr;
-    int *dim_size, *periods, *halosize;
-    int comm_size, comm_rank, comm_size_cart, comm_rank_cart;
-    char *sendbuf, *recvbuf, inputbuf[1000], *pinputbuf;
-    double start, stop, deltatmin, deltatmax, ttt;
-    int rank_source, rank_dest, i, j;
+#define NCALLS 10000
 
-    if (argc>1){
-        printf("%s < ndims dim1 dim2 ... halosize1 halosize2 ...\n", argv[0]);
-        exit(0);
+int
+main (int argc, const char *argv[])
+{
+  MPI_Comm cart_comm, red_comm;
```

Comment:
Which formatting style did you use? One-space identation is... not an identation actually.